### PR TITLE
Feature/binary features

### PR DESCRIPTION
**Pull request by EugKar**
* Enabling optional phonological distinctive feature outputs.
* Enabling hybrid IPA + phonological features mode: using cross-entropy loss for binary feature outputs, then decoding binary features outputs into nearest phoneme, which is used (after converting again into a 
"refined" binary features vector) as the input to the decoder at the next step.
* Leaving diphthongs, affricates, diacritics etc. while preparing IPA targets.

### CODE REVIEW

File: las/model.py [hunk 1]
```diff
@@ -204,23 +216,31 @@ def embedding_fn(ids):
...
             target_embedding = tf.get_variable(
                 'target_embedding', [
-                    hparams.target_vocab_size, hparams.embedding_size],
-                dtype=tf.float32, initializer=tf.contrib.layers.xavier_initializer())
+                    hparams.target_vocab_size,
+                    hparams.embedding_size],
+                dtype=tf.float32, initializer=tf_contrib.layers.xavier_initializer())
 
             return tf.nn.embedding_lookup(target_embedding, ids)
+        elif hparams.binary_outputs:
```
[the remark this answers is not in the document]
Comment (EugKar) [reply las/model.py]: Agree, we should rule it out in parameters validation.

File: las/model.py [hunk 2]
```diff
@@ -236,38 +256,62 @@ def embedding_fn(ids):
...
+    elif hparams.binary_outputs and binf_embedding is None:
+        start_inputs = tf.concat(
+            (tf.zeros([batch_size, hparams.target_vocab_size - 2]),
+             tf.ones([batch_size, 1]),
+             tf.zeros([batch_size, 1])),
+             axis=1
+            )
+        helper = tf_contrib.seq2seq.InferenceHelper(
+            sample_fn=lambda x: tf.round(tf.sigmoid(x)),
+            # sample_fn=lambda x: tf.sigmoid(x),    #TODO: experiment with non-binarlized outputs
```
[the remark this answers is not in the document]
Comment (EugKar) [reply las/model.py]: TODO tells that we may experiment using plain posteriors for the next input instead of Bernoulli samples. In this case, we should also switch off rounding/samplling at training at `training_helper.py`. This is a way of propagating uncertainty to the next step.
As for sampling vs rounding, here (at inference stage without beam search) rounding would lead to deterministic model outputs. Sampling makes sense at training.

File: utils/dataset_utils.py [hunk 3]
```diff
@@ -30,81 +31,121 @@ def parse_fn(example_proto):
...
+        if not is_infer:
+            dataset = dataset.shuffle(output_buffer_size)
+
+        if not binary_targets:
+            dataset = dataset.map(
+                lambda inputs, labels: (inputs,
+                                        vocab_table.lookup(labels)),
+                num_parallel_calls=num_parallel_calls)
+
+        if means is not None and stds is not None:
```
[the remark this answers is not in the document]
Comment (EugKar) [reply utils/dataset_utils.py]: Agree

[the remark this answers is not in the document]
Comment (EugKar) [reply utils/dataset_utils.py]: The code in these two branches differs only by the number of lambda's arguments - there are no labels in the first one.
